openfisca_uk/variables/hmrc/benefit/child_benefit.py

```diff
@@ -0,0 +1,77 @@
+from openfisca_uk.model_api import *
+
+
+class child_benefit_reported(Variable):
+    label = "Child Benefit (reported amount)"
+    documentation = "Reported amount received for Child Benefit"
+    entity = Person
+    definition_period = YEAR
+    value_type = float
+    unit = "currency-GBP"
+
+
+class would_claim_child_benefit(Variable):
+    label = "Would claim Child Benefit"
+    documentation = (
+        "Whether this benefit unit would claim Child Benefit if eligible"
+    )
+    entity = BenUnit
+    definition_period = YEAR
+    value_type = bool
+
+    def formula(benunit, period, parameters):
+        claims_benefits = benunit("claims_all_entitled_benefits", period)
+        takeup_rate = parameters(period).hmrc.benefit.child_benefit.takeup_rate
+        imputed_takeup = (
+            benunit("benunit_random_number", period) <= takeup_rate
+        )
+        return claims_benefits | imputed_takeup
+
+
+class child_benefit_respective_amount(Variable):
+    label = "Child Benefit (respective amount)"
+    documentation = "The amount of this benefit unit's Child Benefit which is in respect of this person"
+    entity = Person
+    definition_period = YEAR
+    value_type = float
+    unit = "currency-GBP"
+
+    def formula(person, period, parameters):
+        eligible = person("is_child_or_QYP", period)
+        is_eldest = person("is_eldest_child", period)
+        child_benefit = parameters(period).hmrc.benefit.child_benefit.amount
+        amount = where(
+            is_eldest, child_benefit.eldest, child_benefit.additional
+        )
+        return eligible * amount * WEEKS_IN_YEAR
+
+
+class child_benefit(Variable):
+    label = "Child Benefit"
+    documentation = "Total Child Benefit for the benefit unit"
+    entity = BenUnit
+    definition_period = YEAR
+    value_type = float
+    unit = "currency-GBP"
+
+    def formula(benunit, period):
+        entitlement = benunit.sum(
```

Comment:
Seems like there's some extra complication here arising from assigning CB to individual children. Do we need to do that? Seems similar to assigning taxes to household heads. Though I see that the reported CB is at the person level.

If we can just implement CB as it's paid (to benunits), we can remove `child_benefit_respective_amount` and `is_eldest_child` and apply this formula instead:
```suggestion
        eligible_children = benunit.sum(benunit.members("is_child_or_QYP", period))
        amount = parameters(period).hmrc.benefit.child_benefit.amount
        eldest_entitlement = amount.eldest * min_(eligible_children, 1)
        additional_entitlement = amount.additional * max_(eligible_children - 1, 0)
        total_entitlement = eldest_entitlement + additional_entitlement
        return benunit("would_claim_child_benefit", period) * total_entitlement
```

Reply:
Yeah I did think about this. But I do favour computing first the child benefit for each child, then adding, because it seems like a good level of decomposition.